CFIN-377 Addition of automated Percy testing (#79)
university-of-york/uoy-app-course-search: README.md

In the first instance developers use Google Chrome Developer Tools and then finally BrowserStack for testing. 
The Wiki page [Testing course search rendering on mobile devices](https://github.com/university-of-york/uoy-app-course-search/wiki/Testing-Course-Search-Rendering-on-Mobile-Devices) has more detail on the process.

#### Percy Automated Testing 

The application has a github workflow action so that at a Pull Request it is subject to automated testing using Percy. 

> Should a difference be detected, then approval is required using Percy prior to a merge. 

Reviewer: Would it be possible to add some instructions on how to approve a difference using Percy?

Author: Yes: I'm writing them for the wiki and also may be presentation? CFIN 394 is that task.

Author: Page added here; https://github.com/university-of-york/uoy-app-course-search/wiki/Testing:-Percy,-automation-in-testing-with-GitHub

Reviewer: Excellent! Would it be possible to link to it from the README?

Author: Added and done

Reviewer 2: That wiki page is great information, but if I'm honest I'm not sure it's necessary :question: 

 For Course Search all we're really interested in is where to find the project and how to approve it - every PR has a check that links straight to the build in Percy, and approval is pretty intuitive, so to be honest I don't think we need to document this beyond what you've put in the README.

The other part of this documentation is how to set up visual testing with Percy. Useful stuff, but Percy documents it pretty well on their website. Perhaps a page in the Confluence wiki might be useful for pointing other teams in the right direction.

Not wanting to be negative because I think this is good information! but I also don't want us to spend time documenting things needlessly, or just because we have a documentation subtask in a ticket.

Author: No problem, I'll move that over into Confluence and leave it at that.

Author: I've moved the page to: https://wiki.york.ac.uk/pages/viewpage.action?pageId=220921899

ESG/ESG Developer Guides/Hints and Tips/Testing: Percy, automation in testing with GitHub.

I'll remove the wiki page.

The Percy project is named "uoy-app-course-search", as uoy-app-course-search is public, so is the Percy project.

> A Percy Badge is added to the bottom of this README to show our use of the package. 

Reviewer: I'm not sure we need to explain that we're adding a badge in the same document where we're adding the badge. This line can probably be removed?

Author: Removed.

In the first instance developers use Google Chrome Developer Tools and then finally BrowserStack for testing. 
The Wiki page [Testing course search rendering on mobile devices](https://github.com/university-of-york/uoy-app-course-search/wiki/Testing-Course-Search-Rendering-on-Mobile-Devices) has more detail on the process.

> #### Percy Automated Testing 

Reviewer 2: ```suggestion
#### Percy Visual Testing 
```

> The application has a github workflow action so that at a Pull Request it is subject to automated testing using Percy. 

Reviewer 2: ```suggestion
The application has a GitHub workflow action so that at a Pull Request it is subject to visual testing using [Percy][Percy](https://percy.io/ad91c322/uoy-app-course-search). 
```

Should a difference be detected, then approval is required using Percy prior to a merge. 

> The Percy project is named "uoy-app-course-search", as uoy-app-course-search is public, so is the Percy project.

Reviewer 2: I think this line can just be removed if we link to the project above as per my other suggestion.

```suggestion
```

## Licence

MIT

> [![This project is using Percy.io for visual regression testing.](https://percy.io/static/images/percy-badge.svg)](https://percy.io/ad91c322/uoy-app-course-search)

Reviewer: Minor thing, but I noticed that [other](https://github.com/university-of-york/esg-app-stutalk-event-bridge) ESG [repositories](https://github.com/university-of-york/ESG-App-MyUoY-Mobile-App) display their badges at the top of the README. I don't know exactly what the convention is, but might be worth considering moving to the top?

Author: Done
